components/Contributors/AddToContributorInsightModal.tsx

```diff
@@ -0,0 +1,169 @@
+      path: `lists/${selectedInsight}/contributors`,
+      body: {
+        contributors: contributors.map((contributor) => {
+          return { login: contributor };
+        }),
+      },
+      bearerToken: sessionToken!,
+      pathValidator: () => safeParse(UuidSchema, selectedInsight).success,
+    });
+
+    if (error) {
+      toast({ description: `Error adding contributors to the insight. Please try again`, variant: "danger" });
+    } else {
+      toast({ description: `Added contributors successfully`, variant: "success" });
+      onCloseModal();
+    }
+  };
+
+  const createContributorInsight = async () => {
+    posthog.capture(`Repo Pages: create new contributor insight`, {
+      repository,
+      count: contributors.length,
+    });
+
+    router.push(
+      `/workspaces/${workspaceId}/contributor-insights/new?contributors=${encodeURIComponent(
+        JSON.stringify(contributors)
+      )}`
+    );
+    return;
+  };
+
+  return (
+    <Dialog open={isOpen}>
+      <DialogContent autoStyle={false} onEscapeKeyDown={onCloseModal} onInteractOutside={onCloseModal}>
+        <Card heading={<h1 className="text-xl font-semibold">Add to workspace</h1>}>
+          <div className="flex flex-col gap-4 w-[32rem] h-full px-8 py-4">
+            {!user ? (
+              <div className="flex flex-col gap-4 text-center">
+                <img
+                  src="/assets/workspace_overview.png"
+                  alt="Workspace screenshot from documentation"
+                  className="border-2 border-light-orange-9 shadow-md rounded-lg"
+                />
+                <Text>
+                  Keep track of repositories and contributors easily with our new feature
+                  <span className="font-semibold"> Workspaces!</span> If you&apos;ve used OpenSauced before, your
+                  insights and lists are now part of your personal workspace.
+                </Text>
+                <p className="font-medium text-light-orange-10">
+                  Create a new contributor insight on a workspace and explore open source like never before!
+                </p>
+                <Button
+                  variant="primary"
+                  className="w-fit gap-2 self-center"
+                  onClick={() => {
+                    signIn({
+                      provider: "github",
+                    });
+                  }}
+                >
+                  <BsGithub className="w-5 h-5" />
+                  Connect with GitHub
+                </Button>
+              </div>
+            ) : (
+              <>
+                <p>Create a new contributor insight or add to an existing one.</p>
+                {isLoading ? (
+                  <p>Loading...</p>
+                ) : (
+                  <SingleSelect
+                    options={[
+                      { label: "Create new insight...", value: "new" },
+                      ...contributorInsights.map(({ id, name }) => ({
```

Comment:
This `contributorInsights` map is stale once you switch to a different workspace:

![Screenshot 2024-07-30 at 2 29 57 PM](https://github.com/user-attachments/assets/628cbfd3-3571-4a6f-b3c4-9589e495d7f7)

Those Contributor Insights are for my default workspace I started with, _not_ for my Cal.com workspace. I'd expect the list of Contributor Insights to update with the selected workspace.

Reply:
When you have 'Create new insight' as the selection, the workspace dropdown shows so you can choose which workspace to create the insight in. The insight list itself is showing all existing insights _throughout all workspaces_.

Maybe I can make the flow easier to understand?

Reply:
Hmmm yeah, it's tricky since there are many contributor insights to many workspaces: I'd suggest first having the user select which workspace they want _and then_ selecting either a new list or an existing list within that workspace. The second dropdown (of selecting a contributor insight) should be greyed out when selecting a workspace so it's clear to the user what they're selecting for. Otherwise, it is confusing knowing which list is in which workspace or where a new list might land - curious if @isabensusan has any opinions on this user flow?

Probably not worth blocking this going in on this either, but we'll definitely want to revisit this user flow since it's abit confusing.

Reply:
I'd say ship as is, and let's review the flow in a follow up PR (opening an Issue #3818 ). Adding stuff to workspaces as a while needs to be revisited tbh